src/feature/nodelist/microdesc.c

```diff
@@ -1057,3 +1057,40 @@ usable_consensus_flavor,(void))
     return FLAV_NS;
   }
 }
+
+/* Return a newly allocated RSA key object from the ASN-1 encoded RSA onion
+ * public key in the microdescriptor. It is the caller responsability to free
+ * the returned object.
+ *
+ * Return NULL if md or the md's onion pkey is NULL or malformed. */
+crypto_pk_t *
+microdesc_get_rsa_onion_pkey(const microdesc_t *md)
```

Comment:
These two functions seem redundant with their routerinfo_t equivalent.  Do you think it's reasonable to combine them -- especially the "set" one, since it's longer?

Reply:
Agrree. I merge them together and put the helper functions in `router.c`. Hope this is fine. See fixup commit: 61bdbe40c7ea5b7d